amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -1,743 +1,322 @@
-        benchmark_api = json.loads if io_type == Io_type.BUFFER.value else json.load
-    elif format_option == Format.SIMPLEJSON.value:
-        benchmark_api = simplejson.loads if io_type == Io_type.BUFFER.value else simplejson.load
-    elif format_option == Format.UJSON.value:
-        benchmark_api = ujson.loads if io_type == Io_type.BUFFER.value else ujson.load
-    elif format_option == Format.RAPIDJSON.value:
-        benchmark_api = rapidjson.loads if io_type == Io_type.BUFFER.value else rapidjson.load
-    elif format_option == Format.CBOR.value:
-        benchmark_api = cbor.loads if io_type == Io_type.BUFFER.value else cbor.load
-    elif format_option == Format.CBOR2.value:
-        benchmark_api = cbor2.loads if io_type == Io_type.BUFFER.value else cbor2.load
-    else:
-        raise Exception(f'unknown JSON/CBOR/Ion format {format_option} to generate setup code.')
-
-    if io_type == Io_type.BUFFER.value:
-        with open(file, 'br') as fp:
-            benchmark_data = fp.read()
-
-        if not memory_profiling:
-            def test_func():
-                data = benchmark_api(benchmark_data)
-                return data
-        else:
-            def test_func():
-                tracemalloc.start()
-                data = benchmark_api(benchmark_data)
-                global read_memory_usage_peak
-                read_memory_usage_peak = tracemalloc.get_traced_memory()[1] / BYTES_TO_MB
-                tracemalloc.stop()
-                return data
-    else:
-        if not memory_profiling:
-            def test_func():
-                with open(file, 'br' if binary else 'r') as benchmark_file:
-                    data = benchmark_api(benchmark_file)
-                return data
-        else:
-            def test_func():
-                tracemalloc.start()
-                with open(file, 'br' if binary else 'r') as benchmark_file:
-                    data = benchmark_api(benchmark_file)
-                global read_memory_usage_peak
-                read_memory_usage_peak = tracemalloc.get_traced_memory()[1] / BYTES_TO_MB
-                tracemalloc.stop()
-                return data
-
-    return test_func
-
-
-# Generates benchmark code for event based API
-def generate_event_test_code(file):
-    pass
-
-
-# Generates setup code for json/cbor benchmark code
-def generate_setup(format_option, c_extension=False, gc=False, memory_profiling=False):
-    if format_is_ion(format_option):
-        rtn = f'import gc; import amazon.ion.simpleion as ion;from amazon.ion.simple_types import IonPySymbol; ' \
-              f'ion.c_ext = {c_extension}'
-    else:
-        rtn = 'import gc'
 
-    if gc:
-        rtn += '; gc.enable()'
-    else:
-        rtn += '; gc.disable()'
+TOOL_VERSION = '1.0.0'
+
+
+def _str_to_bool(v):
+    """
+    Convert a string CLI arg to a boolean
+    """
+    if isinstance(v, str):
+        return v.lower() in ("true", "1")
+    return False
+
```

Comment:
I don't know docopt but I'm surprised this is a concern the user needs to worry about.

Reply:
This function predates this PR—I just moved it. I can check and see how docopt handles booleans.

Reply:
I got rid of it entirely.